tests/State/ScopeTest.php

```diff
@@ -82,7 +77,12 @@ public function testSetExtras(): void
         $this->assertSame(['foo' => 'bar', 'bar' => 'baz'], $event->getExtraContext()->toArray());
     }
 
-    public function testSetUser(): void
+    /**
+     * @group legacy
+     *
+     * @expectedDeprecation Replacing user context on setUser is deprecated since sentry-php 2.3 and will be changed to a merge on 3.0. For now you can use $merge = true to merge instead of replacing it.
+     */
+    public function testDeprecatedSetUser(): void
```

Comment:
Please rename this method to `testSetUserThrowsDeprecation`

Reply:
c3dfd42

Reply:
c3dfd42